rosbag2_transport/test/rosbag2_transport/test_play_topic_remap.cpp

```diff
@@ -32,14 +31,16 @@
 
 #include "rosbag2_play_test_fixture.hpp"
 
-
 TEST_F(RosBag2PlayTestFixture, recorded_message_is_played_on_remapped_topic) {
   //  test constants
   const std::string original_topic = "/topic_before_remap";
   const std::string remapped_topic = "/topic_after_remap";
   const int32_t test_value = 42;
-  const int64_t timestamp_in_milliseconds = 700;
-  const size_t expected_number_of_messages = 1;
+  const int64_t dt_in_milliseconds = 100;
+  // With ROS2 communication being undeterministic, we can only verify that topics arrive
+  // We can't verify how many though; The 60% below seem to be reliable.
```

Comment:
Does this count really matter for this test, or should we be happy to see just a single message arrive? I would think that a more reliable test would be to publish an arbitrarily large number of messages and interrupt execution when we receive a single message on the subscription. (IIRC we don't have a good way to stop playback yet, though, so maybe the point is just "we should expect 1 message")

Reply:
I was thinking about this. And while I am not settled on the 60% either, I was arguing with myself to apply the "1, 2, many" to rule out there're no leftovers in the system. That's not to say that this is reasonable in the first place :P So I am happy to modify the test in a way to expect at least 1 message.

Reply:
I think that's the right thing to do for now. It's unfortunate that we can't depend on the specific numbers, but that's not the specific thing being tested here so we might as well loosen the restrictions all the way to only require the exact behavior being tested to work.